packages/grafana-data/src/types/logs.ts

```diff
@@ -194,6 +195,35 @@ export enum LogsVolumeType {
   Limited = 'Limited',
 }
 
+/**
+ * Custom meta information required by Logs Volume responses
+ */
+export type LogsVolumeCustomMetaData = {
+  absoluteRange: AbsoluteTimeRange;
+  logsVolumeType: LogsVolumeType;
+  datasourceName: string;
+  sourceQuery: DataQuery;
+};
+
+export const getLogsVolumeAbsoluteRange = (
+  dataFrames: DataFrame[],
+  defaultRange: AbsoluteTimeRange
+): AbsoluteTimeRange => {
+  return dataFrames[0].meta?.custom?.absoluteRange || defaultRange;
+};
+
+export const getLogsVolumeDataSourceInfo = (dataFrames: DataFrame[]): { uid: string; name: string; refId: string } => {
```

Comment:
i know it's probably never going to happen, but maybe doing something like
```
export const getLogsVolumeDataSourceInfo = (
  dataFrames: DataFrame[]
): { uid: string; name: string; refId: string } | null => {
  if (
    dataFrames[0]?.meta?.custom?.datasourceUid &&
    dataFrames[0]?.meta?.custom?.datasourceName &&
    dataFrames[0]?.meta?.custom?.refId
  ) {
    return {
      uid: dataFrames[0]?.meta?.custom?.datasourceUid || '',
      name: dataFrames[0]?.meta?.custom?.datasourceName || '',
      refId: dataFrames[0]?.meta?.custom?.sourceQuery?.refId || '',
    };
  } else {
    return null;
  }
};
```

would be a bit safer to avoid returning empty string that are not handled by the logs volume panel? wdyt?

Reply:
Makes sense! And while fixing it I also noticed that uid is not used any more so I removed it. Updated in 7ab13b64807dcaa2230ff3a94aacfed477683945